### ISSUE TITLE
Option for Donut Pie Chart

### ISSUE THREAD

**Issue by kikislater** [ghtmtt/DataPlotly#231]
**Describe the solution you'd like**
I succeed to represent donut pie chart like in this map : 

![image](https://user-images.githubusercontent.com/13238064/82550997-76344180-9b70-11ea-9a43-9beafc14475e.png)
Now I think it could be benefit to others to have such option in this plugin. 

**Describe alternatives you've considered**
Like in documentation, I just add **hole=.3,** in core/plot_type/pie.py

https://plotly.com/python/pie-charts/#donut-chart

But better implementation and controlling the size of the hole should be nice. So what is needed to do (please correct me if I'm wrong) ? 
- declare option in core/plot_settings.py
- declare option in gui/plot_settings_widget.py
- Add option to .ui

**Additional context**
URL and screen above

**Comment by dominijk** [comment]
@kikislater  Where can I find the code `pie.py` to edit this?

**Comment by kikislater** [comment]
in core/plot_type/pie.py =>
https://github.com/ghtmtt/DataPlotly/blob/master/DataPlotly/core/plot_types/pie.py#L42-L43

**Comment by ghtmtt** [comment]
Thanks @kikislater for helping. @dominijk here are other instructions. Please don't hesitate to add comments if something is not clear. 

You should also add an eventual checkbox to the UI (here https://github.com/ghtmtt/DataPlotly/blob/master/DataPlotly/gui/plot_settings_widget.py#L539) and then be sure to add the widget to the correct plot type during the UI refresh (here https://github.com/ghtmtt/DataPlotly/blob/master/DataPlotly/gui/plot_settings_widget.py#L539).

Finally, depending on the widget and its value, you should pass the parameter to the function https://github.com/ghtmtt/DataPlotly/blob/master/DataPlotly/gui/plot_settings_widget.py#L539

**Comment by dominijk** [comment]
@ghtmtt and @kikislater thank you, so i'd need to install the plugin from my own build rather than just find the python file from the official plugin and edit that?

**Comment by ghtmtt** [comment]
> @ghtmtt and @kikislater thank you, so i'd need to install the plugin from my own build rather than just find the python file from the official plugin and edit that?

yeah, that you be definitely better so that you can test locally if the changes introduced are working nice

**Comment by kikislater** [comment]
> You should also add an eventual checkbox to the UI (here https://github.com/ghtmtt/DataPlotly/blob/master/DataPlotly/gui/plot_settings_widget.py#L539) and then be sure to add the widget to the correct plot type during the UI refresh (here https://github.com/ghtmtt/DataPlotly/blob/master/DataPlotly/gui/plot_settings_widget.py#L539).
> 
> Finally, depending on the widget and its value, you should pass the parameter to the function https://github.com/ghtmtt/DataPlotly/blob/master/DataPlotly/gui/plot_settings_widget.py#L539

Every modifications are in the same function ? All links are the same.

**Comment by ghtmtt** [comment]
> Every modifications are in the same function ? All links are the same.

my fault, comment edited

**Comment by kikislater** [comment]
Thank you Matteo !
